android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/NearbyBeaconsFragment.java

```diff
@@ -222,7 +228,17 @@ public void onResume() {
     getActivity().getActionBar().setTitle(R.string.title_nearby_beacons);
     getActivity().getActionBar().setDisplayHomeAsUpEnabled(false);
     getListView().setVisibility(View.INVISIBLE);
-    mDiscoveryServiceConnection.connect(true);
+    if (mFirstTime && !mMainActivity.mCheckingPermissions) {
```

Comment:
Could you create an accessor method in MainActivity so we don't have to access mCheckingPermissions directly?  Then mCheckingPermissions can be made private again.